azure-pipelines/main-pipeline.yml

```diff
@@ -119,33 +135,38 @@ stages:
           cliProjectKey: '$(sonarProjectKey)'
           cliProjectName: '$(sonarProjectName)'
           cliSources: '.'
-          extraProperties: 'sonar.java.binaries=**/target/classes'
+          extraProperties: |
+              sonar.java.binaries=**/target/classes
+              sonar.exclusions=/jdk/**/*
```

Comment:
1. isn't that /jdk/ path different from the path that your jdk step above (& below) is using? or are those not related?
2. be consistent about indentation.
3. if these sonar steps are analyzing all of the services, clarify that with a brief comment
4. is it possible that variables['base'] is false, but we still have changes that we want to be analyzed with sonar but that this pipeline logic will skip?

Reply:
I believe I've taken care of all of this.